<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <comment>
********************************************************************************
**
** Copyright (c) 2017 The University of Notre Dame
** Copyright (c) 2017 The Regents of the University of California
**
** Redistribution and use in source and binary forms, with or without modification,
** are permitted provided that the following conditions are met:
**
** 1. Redistributions of source code must retain the above copyright notice, this 
** list of conditions and the following disclaimer.
**
** 2. Redistributions in binary form must reproduce the above copyright notice, this
** list of conditions and the following disclaimer in the documentation and/or other
** materials provided with the distribution.
**
** 3. Neither the name of the copyright holder nor the names of its contributors may
** be used to endorse or promote products derived from this software without specific
** prior written permission.
**
** THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS &quot;AS IS&quot; AND ANY
** EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES
** OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT
** SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
** INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED
** TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR
** BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
** CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING
** IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
** SUCH DAMAGE.
**
**********************************************************************************

// Contributors:
// Written by Peter Sempolinski, for the Natural Hazard Modeling Laboratory, director: Ahsan Kareem, at Notre Dame
 </comment>
 <class>ExplorerWindow</class>
 <widget class="QMainWindow" name="ExplorerWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1000</width>
    <height>700</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Agave Explorer</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <widget class="HeaderWidget" name="header">
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Raised</enum>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QTabWidget" name="stackedView">
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="filePage">
       <attribute name="title">
        <string>Remote Files</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <item>
         <widget class="QLabel" name="fileSelectedHeader">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Minimum" vsizetype="Maximum">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="minimumSize">
           <size>
            <width>200</width>
            <height>0</height>
           </size>
          </property>
          <property name="font">
           <font>
            <pointsize>14</pointsize>
           </font>
          </property>
          <property name="text">
           <string>Selected File:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="SelectedFileLabel" name="selectedFileInfo">
          <property name="minimumSize">
           <size>
            <width>0</width>
            <height>150</height>
           </size>
          </property>
          <property name="text">
           <string>No File Selected.</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="filesLabel">
          <property name="text">
           <string>Remote Files:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="RemoteFileTree" name="remoteFileView">
          <property name="contextMenuPolicy">
           <enum>Qt::CustomContextMenu</enum>
          </property>
          <property name="editTriggers">
           <set>QAbstractItemView::NoEditTriggers</set>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab">
       <attribute name="title">
        <string>Agave Jobs</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_3">
        <item>
         <widget class="QLabel" name="label_2">
          <property name="text">
           <string>Agave Job List:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="RemoteJobLister" name="jobTable">
          <property name="contextMenuPolicy">
           <enum>Qt::CustomContextMenu</enum>
          </property>
          <property name="editTriggers">
           <set>QAbstractItemView::NoEditTriggers</set>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="AgavePassThru">
       <attribute name="title">
        <string>Agave Apps</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout">
        <item row="3" column="1" colspan="2">
         <widget class="QPushButton" name="agaveAppEnactButton">
          <property name="text">
           <string>Enact Agave App</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QLabel" name="label">
          <property name="text">
           <string>File Selected:</string>
          </property>
         </widget>
        </item>
        <item row="0" column="0" rowspan="5">
         <widget class="QListView" name="agaveAppList">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
            <horstretch>1</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="editTriggers">
           <set>QAbstractItemView::NoEditTriggers</set>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="SelectedFileLabel" name="selectedFileLabel">
          <property name="text">
           <string>None</string>
          </property>
         </widget>
        </item>
        <item row="4" column="1">
         <widget class="QScrollArea" name="ParamArea">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
            <horstretch>2</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="widgetResizable">
           <bool>true</bool>
          </property>
          <widget class="QWidget" name="AgaveParamWidget">
           <property name="geometry">
            <rect>
             <x>0</x>
             <y>0</y>
             <width>630</width>
             <height>518</height>
            </rect>
           </property>
           <layout class="QGridLayout" name="gridLayout_2"/>
          </widget>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
    <item>
     <widget class="FooterWidget" name="footer">
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Raised</enum>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <action name="actionExit">
   <property name="text">
    <string>Exit</string>
   </property>
  </action>
  <action name="actionCopyInfo">
   <property name="text">
    <string>License Info</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>RemoteFileTree</class>
   <extends>QTreeView</extends>
   <header>remoteModelViews/remotefiletree.h</header>
  </customwidget>
  <customwidget>
   <class>HeaderWidget</class>
   <extends>QFrame</extends>
   <header>../AgaveExplorer/SimCenterCommon/HeaderWidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>FooterWidget</class>
   <extends>QFrame</extends>
   <header>../AgaveExplorer/SimCenterCommon/FooterWidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>RemoteJobLister</class>
   <extends>QTableView</extends>
   <header>remoteModelViews/remotejoblister.h</header>
  </customwidget>
  <customwidget>
   <class>SelectedFileLabel</class>
   <extends>QLabel</extends>
   <header>remoteModelViews/selectedfilelabel.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>agaveAppList</sender>
   <signal>clicked(QModelIndex)</signal>
   <receiver>ExplorerWindow</receiver>
   <slot>agaveAppSelected(QModelIndex)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>177</x>
     <y>363</y>
    </hint>
    <hint type="destinationlabel">
     <x>499</x>
     <y>349</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>agaveAppEnactButton</sender>
   <signal>clicked()</signal>
   <receiver>ExplorerWindow</receiver>
   <slot>agaveCommandInvoked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>660</x>
     <y>105</y>
    </hint>
    <hint type="destinationlabel">
     <x>499</x>
     <y>349</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>setTestVisual()</slot>
  <slot>setMeshVisual()</slot>
  <slot>agaveAppSelected(QModelIndex)</slot>
  <slot>agaveCommandInvoked()</slot>
 </slots>
</ui>
